pkg/decision/composite_service.go

```diff
@@ -113,6 +114,9 @@ func (s CompositeService) GetFeatureDecision(featureDecisionContext FeatureDecis
 				convertedValue, e = strconv.ParseFloat(variableValue, 64)
 			case entities.Boolean:
 				convertedValue, e = strconv.ParseBool(variableValue)
+			case entities.JSON:
+				convertedValue = map[string]string{}
+				e = json.Unmarshal([]byte(variableValue), &convertedValue)
```

Comment:
Should we pass the OptimizelyJSON object to the notification, rather than the JSON string? And apply memoization to creating the OptimizelyJSON as @mikeng13 suggested?

Reply:
memoization is already done on OptimizelyJSON, the thing is that GetFeatureDecision does not know anything about OptimizelyJSON. This whole notification logic will need to be restructured.